src/docserv/functions.py

```diff
@@ -32,6 +33,40 @@ def mail(text, subject, to):
     p.communicate(msg.as_string())
 
 
+def file_paths(directory, zip_formats):
+    """Collect all files with paths in specified formats from given directory.
+
+    :param str directory: a directory from which files are to be collected
+    :param list zip_formats: a list containing accepted documentation formats
+        example: ["pdf", "epub", "single-html"]
+    :return: a generator of matching file paths
+    """
+    for root, directories, files in os.walk(directory):
+        for filename in files:
+            filepath = os.path.join(root, filename)
+            for format in zip_formats:
+                if (format in root) or (root == directory and format in filepath):
+                    yield filepath
+                    break
+
+def create_zip_file(directory, zip_formats, zip_file):
```

Comment:
Afaik, this function is not yet called by anything?

Reply:
nope